codegen/templates/code/types/clients.gotmpl

```diff
@@ -129,10 +129,10 @@ type {{ $resource.Kind }}Writer interface {
 type {{ $resource.Kind }}StatusWriter interface {
     // Update updates the fields corresponding to the status subresource for the
     // given {{ $resource.Kind }} object.
-    Update{{ $resource.Kind }}Status(ctx context.Context, obj *{{ $import_prefix }}{{ $resource.Kind }}, opts ...client.UpdateOption) error
+    Update{{ $resource.Kind }}Status(ctx context.Context, obj *{{ $import_prefix }}{{ $resource.Kind }}, opts ...client.SubResourceUpdateOption) error
```

Comment:
do you know why these had to be renamed? as far as i can tell, `UpdateOption` and `PatchOption` both still exist in controller-runtime. do they behave differently in the newest controller-runtime (e.g., are they enforcing that we call the 'subresource' methods on statuses now)?

Reply:
From https://github.com/kubernetes-sigs/controller-runtime/pull/2072
> Use distinct options for subresource Update and Patch which makes this a breaking change

So I can't comment on _why_ these API breakages had to happen other than link to the upstream pull request, but I can say that while `UpdateOption` and PatchOption` still exist, but now can only be applied to resources that aren't subresources.